src/transport/mqtt/pairing.rs

```diff
@@ -90,28 +88,32 @@ pub enum PairingError {
     },
     /// Couldn't read native certificates
     #[error("couldn't read native certificates")]
-    ReadNativeCerts(#[from] tokio::task::JoinError),
+    ReadNativeCerts(#[source] tokio::task::JoinError),
 }
 
 /// Struct with the information for the pairing
 pub(crate) struct ApiClient<'a> {
     pub(crate) realm: &'a str,
     pub(crate) device_id: &'a str,
-    pairing_url: &'a Url,
-    credentials_secret: &'a str,
+    pairing_url: Url,
+    credentials_secret: String,
+    tls: ClientConfig,
```

Comment:
What do you think about call it `tls_config`?
```suggestion
    tls_config: ClientConfig,
```

Reply:
LGTM